services/wallet/gemini_test.go

```diff
@@ -0,0 +1,207 @@
+package wallet
+
+import (
+	"testing"
+
+	"github.com/brave-intl/bat-go/libs/clients/gemini"
+	should "github.com/stretchr/testify/assert"
+)
+
+func Test_getIssuingCountry(t *testing.T) {
+	type tcGiven struct {
+		gx       *geminix
+		validAcc gemini.ValidatedAccount
+		fallback bool
+	}
+
+	tests := []struct {
+		name     string
+		given    tcGiven
+		expected string
+	}{
+		{
+			name: "has_prior_linking_no_valid_documents",
+			given: tcGiven{
+				gx: newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{
+					CountryCode: "US",
+				},
+				fallback: true,
+			},
+			expected: "US",
+		},
+		{
+			name: "has_prior_linking_and_valid_documents",
+			given: tcGiven{
+				gx: newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{
+					CountryCode: "US",
+					ValidDocuments: []gemini.ValidDocument{
+						{
+							Type:           "passport",
+							IssuingCountry: "PT",
+						},
+					},
+				},
+				fallback: true,
+			},
+			expected: "PT",
+		},
+		{
+			name: "has_no_prior_linking_and_no_valid_documents",
+			given: tcGiven{
+				gx: newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{
+					CountryCode: "US",
+				},
+				fallback: false,
+			},
+			expected: "",
+		},
+		{
+			name: "has_no_prior_linking_and_valid_documents",
+			given: tcGiven{
+				gx: newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{
+					CountryCode: "US",
+					ValidDocuments: []gemini.ValidDocument{
+						{
+							Type:           "passport",
+							IssuingCountry: "PT",
+						},
+					},
+				},
+				fallback: false,
+			},
+			expected: "PT",
+		},
+		{
+			name: "has_prior_linking_and_no_country_code_and_no_valid_documents",
+			given: tcGiven{
+				gx:       newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{},
+				fallback: false,
+			},
+			expected: "",
+		},
+		{
+			name: "has_prior_linking_and_no_country_code_and_valid_documents",
+			given: tcGiven{
+				gx: newGeminix("passport"),
+				validAcc: gemini.ValidatedAccount{
+					ValidDocuments: []gemini.ValidDocument{
+						{
+							Type:           "passport",
+							IssuingCountry: "PT",
+						},
+					},
+				},
+				fallback: true,
+			},
+			expected: "PT",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := tc.given.gx.GetIssuingCountry(tc.given.validAcc, tc.given.fallback)
+			should.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func Test_countryForDocByPrecedence(t *testing.T) {
```

Comment:
Nit: Since we're testing a function, then the test name should be without the underscore – `TestCountryForDocByPrecedence`.

```suggestion
func TestCountryForDocByPrecedence(t *testing.T) {
```